### PR TITLE
Mod- and UI- related stuff

### PR DESCRIPTION
**Pull request by aimozg**

### CODE REVIEW

File: content/coc/mountain/hellhound.xml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+		<else/>
+		<t>His flaming seed brings about a pleasure you had not expected; your insides feel like they are alight with passion and power. You feel a similar joy on your back, and your muscles feel like they are being massaged from the heat.  It is an incredible and fiery experience, one that you don't think you could have had if it wasn't for the power of corruption that you've gained since you got here. Too soon, the beast pulls out of you, giving your [asshole] a single extended lick with its long dog-like tongue before running off. You quickly look over and manage to catch a glimpse of its tail before it disappears from view, just before your body falls into a deep sleep. When you wake, you can still feel the afterglow of the flaming seed inside you.</t>
+	</if>
+</if>
+<command>Orgasm("Generic")</command>
```
Comment (Oxdeception) [inline content/coc/mountain/hellhound.xml:66]:
```suggestion
<command>Orgasm(player, "Generic")</command>
```

File: content/coc/mountain/hellhound.xml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	<t>One of the heads eagerly starts licking your [cock] while the other starts to work on your </t>
+	<output>cockDescript(1)</output>
+	<t>. The large, flat tongues give you a rather unusual feeling, but it is definitely effective.   As you approach your peak, the heads quickly take your [cock] and </t>
+	<output>cockDescript(1)</output>
+	<t> into their mouths. The pleasure from this new arrangement quickly overwhelms you, leaving you dazed.  When your senses return to you, you look up to see the hellhound's tongues running all over their mouths, clearly very happy.  With a pair of happy barks, the hellhound turns and runs away.</t>
+</if>
+<if test="Sex() == 'N'">
+	<t>The two heads stare at your lack of sexual endowments for a few seconds before looking up at you sadly.  With a whine, the hellhound wanders off.  Feeling rather unsatisfied, you think that you should probably fix your lack of sexual 'parts'...</t>
+	<else/>
+	<command>Orgasm("Generic")</command>
```
Comment (Oxdeception) [inline content/coc/mountain/hellhound.xml:94]:
```suggestion
	<command>Orgasm(player, "Generic")</command>
```

File: classes/coc/xxc/StdFunctions.as [hunk 3]
```diff
@@ -3,17 +3,21 @@
...
+import classes.internals.EnumValue;
 
 public class StdFunctions {
-	public function CreatureHasPerk(creature:Creature,has:Boolean,perkId:String):Boolean {
-		return creature.hasPerk(PerkType.lookupPerk(perkId)) == has;
-	}
-	public function CreatureHasTrait(creature:Creature,trait:String,invert:Boolean):Boolean {
-		if (invert) return !CreatureHasTrait(creature,trait,false);
+
+	public function CreatureHasTrait(creature:Creature,trait:String):Boolean {
```
Comment (Oxdeception) [inline classes/coc/xxc/StdFunctions.as:35]: Mod XML needs updating to account for change in parameters. Erlking/Goblin

File: classes/coc/xxc/StdFunctions.as [hunk 4]
```diff
@@ -22,12 +26,46 @@ public class StdFunctions {
 			case 'horns': return creature.horns.type != Horns.NONE;
 			case 'tails': return creature.tail.type != Tail.NONE;
 			default:
-				trace("[WARN] CreatureHasTrait('"+trait+"')");
+				warn("CreatureHasTrait",trait);
 				return false;
 		}
 	}
-	public function CreatureTestSex(creature:Creature,sexTest:/*int*/Array):Boolean {
```
Comment (Oxdeception) [inline classes/coc/xxc/StdFunctions.as:29]: Need to update bundled mod XML's to account for this being removed.

**Review by Oxdeception — CHANGES_REQUESTED** [review]
Please also add some form of background to the Module Menu, to make more easily read.
Could do this by allowing changes to the CoCScrollPane background alpha

**Comment by aimozg** [comment]
My bad, missed some of Std(Commands/Functions) changes.
Fixed, also added simple validator (checks arg counter for simple commands and tests)

**Comment by aimozg** [comment]
@Oxdeception All done

**Pull request merged**